### ISSUE TITLE
If `files.assets === 'public'`, Vite copies the directory's contents

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/kit#587]
This potentially conflicts with our own static asset handling

**Comment by benmccann** [comment]
Not sure if it's the same issue, but I saw @GrygrFlzr filed https://github.com/vitejs/vite/issues/2587 about Vite's public directory as well

**Comment by jthegedus** [comment]
Hi, this issue is the cause for an annoying workflow/bug with the [Firebase adapter](https://github.com/jthegedus/svelte-adapter-firebase/issues/22).

The adapter reads the user's `firebase.json` configuration file to identify which dir contains static assets to push to the Firebase CDN, in Firebase this defaults to `public` and is almost never changed by the user. This results in the following process:

- `svelte-kit build` outputs `.svelte/output/client/_app/*`
- Firebase adapter then executes `copy_static_files` and `copy_client_files` which copies the `static` dir and `.svelte/output/client/_app/*` dir contents to `public`
- on subsequent `svelte-kit build` executions Vite then copies `public` contents into `.svelte/output/client/_app/*`, then outputs the new build as well, resulting in .svelte/output/client/_app/*` and .svelte/output/client/_app/_app/*`. This then gets copied by the adapter `copy_static_files` and `copy_client_files` into `public`.
- each re-run of `svelte-kit build` produces a further nested `_app` dir

Is it possible for my adapter to specifically disable this Vite config or should we disable it within SvelteKit, because it is causing a conflict for my use case.

**Comment by jthegedus** [comment]
The current workaround for the [Firebase adapter is documentation](https://github.com/jthegedus/svelte-adapter-firebase#quickstart) to users to add

```
"build": "npx -y rimraf <dir used in firebase.json.hosting.public> && svelte-kit build --verbose",
```

to their `package.json`. Again, the default value is `public`, but ideally this would be something these users didn't have to worry about.

**Comment by theodorejb** [comment]
I am also running into this issue. Is there an official workaround or fix in the works?

**Comment by theodorejb** [comment]
I was able to work around this issue by setting the following in `svelte.config.js`:

```js
const config = {
    // ...
    kit: {
        // ...
        vite: {
            publicDir: false,
        },
    },
};
```

Would it make sense for SvelteKit to set `publicDir` to `false` by default to avoid the current unexpected behavior?

**Comment by bluepuma77** [comment]
Shouldn't this be resolved in latest `SvelteKit v1.0.0-next.295` from npm?

When I place static files in SvelteKit `/static/public` I get the following message:
```
files in the public directory are served at the root path.
Instead of /public/bootstrap/bootstrap.min.css, use /bootstrap/bootstrap.min.css.
```
I still have to manually set the vite setting and restart SvelteKit.
```
const config = {
  kit: {
    ...
    vite: {
      publicDir: false
    }
  }
};
```
